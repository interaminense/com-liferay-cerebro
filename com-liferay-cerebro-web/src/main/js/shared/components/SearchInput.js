import autobind from 'autobind-decorator';
import Component, {Config} from 'metal-jsx';
import {noop} from 'lodash';

import Button from './Button';
import Icon from './Icon';
import Input from './Input';

class SearchInput extends Component {
	@autobind
	handleKeyDown(event) {
		if (event.key === 'Enter') {
			this.props.onSubmit(this.refs.input.element.value);
		}
	}

	@autobind
	handleSubmit() {
		this.props.onSubmit(this.refs.input.element.value);
	}

	render() {
		const {placeholder} = this.props;

		return (
			<div class={this.context.navBar ? 'navbar-form' : ''}>
				<Input.Group inset>
					<Input.GroupInput>
						<Input
							{...this.otherProps()}
							onKeyDown={this.handleKeyDown}
							placeholder={placeholder}
							ref="input"
						/>
					</Input.GroupInput>

					<Input.Inset>
						<Button display="unstyled" onClick={this.handleSubmit}>
							<Icon symbol="search" />
						</Button>
					</Input.Inset>
				</Input.Group>
			</div>
		);
	}
}

SearchInput.PROPS = {
	onSubmit: Config.func().value(noop),
	placeholder: Config.string().value(Liferay.Language.get('search'))
};

export default SearchInput;