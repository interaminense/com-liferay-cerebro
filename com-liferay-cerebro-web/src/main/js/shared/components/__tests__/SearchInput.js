import SearchInput from '../SearchInput';

describe('SearchInput', () => {
	let component;

	afterEach(() => {
		if (component) {
			component.dispose();
		}
	});

	it('should render input', () => {
		component = new SearchInput();

		expect(component).toMatchSnapshot();
	});

	it('should return input value if key is `Enter`', () => {
		const onSubmit = jest.fn();
		const value = 'foo';

		component = new SearchInput({onSubmit, value});

		component.handleKeyDown({key: 'Enter'});

		expect(onSubmit).toHaveBeenCalledWith(value);
	});

	it('should return input value', () => {
		const onSubmit = jest.fn();
		const value = 'foo';

		component = new SearchInput({onSubmit, value});

		component.handleSubmit();

		expect(onSubmit).toHaveBeenCalledWith(value);
	});
});