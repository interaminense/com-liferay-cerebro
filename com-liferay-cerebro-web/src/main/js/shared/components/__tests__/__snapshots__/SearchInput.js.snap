// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput should render input 1`] = `
<div class="">
  <div class="input-group input-group-inset">
    <div class="input-group-input">
      <input placeholder="search" class="input-root form-control" type="text">
    </div>
    <span class="input-group-inset-item">
      <button class="button-root btn btn-unstyled" type="button">
        <svg class="icon-root lexicon-icon lexicon-icon-search">
          <use xlink:href="/images/lexicon/icons.svg#search"></use>
        </svg>
      </button>
    </span>
  </div>
</div>
`;
